### ISSUE TITLE
Add unit tests

### ISSUE THREAD

**Issue by Ejhfast** [CityOfZion/neon-wallet#34]
Neon-js has tests, but the wallet does not. Should be easy enough to add tests for the reducer logic and helper methods. UI/DOM oriented tests would be great, but maybe next step

**Comment by dvdschwrtz** [comment]
going to start looking though the code and plan on writing some additional tests later this week or next.

**Comment by Alexintosh** [comment]
I'd like to help with this.
In order to make the process easier I'd like to suggest a discussion about the architecture of the components.

For instance:
- Use [container/component architecture](https://medium.com/@dan_abramov/smart-and-dumb-components-7ca2f9a7c7d0). (right now many components use the connect function).
- Embrace component encapsulation with styled-components

**Comment by dvdschwrtz** [comment]
I agree that any components using connect should be split out. However, I think we should just start by writing tests and move that to another issue. I also think that the styled-components could be interesting. Just read about it. However, I also think that should be a separate issue. The snapshot tests can cover any style changes and the tests should still work if we restructure the folders.

I still need to look through the app... Its been a crazy week, but I am hoping to get started on writing some tests this weekend. Any help is greatly appreciated. And please let me know if you disagree with the points I made above.

**Comment by Alexintosh** [comment]
I'm okay in moving the improvements into a separate issue, I will open it right away.
My only concern in writing the tests before changing the architecture is that we will probably have to  rewrite some tests once the migration is done.

About the styled-components, glad you liked it :) 
I've been working with these for quite a while now in large react project, I believe it's a good improvement for complete modularity and scalability of the application.

**Comment by dvdschwrtz** [comment]
My first open source project... I see there have been some PR for unit tests already. Is there a way to prevent overlap, so that I don't work on the same tests as someone else?

Should I just make an empty commit and PR with the name of the tests I am writing and keep pushing to that and then assign it when it's ready? Also, am I supposed to rebase my work before I create a merge request?

**Comment by dvdschwrtz** [comment]
writing tests for TransactionHistory component right now FYI for anyone writing tests... hate the idea of overlap for some reason :)

**Comment by danielfm123** [comment]
You can use that neo web wallet https://www.myneowallet.io

**Comment by dvdschwrtz** [comment]
For anyone looking to write tests, this article is very helpful in understanding what is happening with enzyme’s shallow rendering (http://brewhouse.io/2016/03/18/accelerate-your-react-testing-with-enzyme.html).

**Comment by phetter** [comment]
FYI, I've created a unit test for LoginLocalStorage as part of https://github.com/CityOfZion/neon-wallet/pull/237 and will continue to increase coverage in further PRs. That module is at about 42% coverage as of the aforementioned PR.

**Comment by mhuggins** [comment]
@phetter I'm looking to start contributing to CoZ repos for the first time this week.  Is this something you need help with, or should I pick up something else?
